tests/test_processing.py

```diff
@@ -392,3 +448,381 @@ def get_all(file_name, report_type: Type[T]) -> List[T]:
+):
+    process_result = process_file(multi_file_extraction_config, multi_volume_zip)
+    task_result_by_path = {r.task.path: r for r in process_result.results}
+
+    data = extraction_root / "input_extract" / "test_extract" / "data"
+
+    out = (
+        extraction_root
+        / "input_extract"
+        / "test_extract"
+        / "data_extract"
+        / "0-1.dummy_extract"
+        / "AA"
+    )
+
+    assert out.exists()
+    assert out.read_bytes() == b"A"
+
+    assert task_result_by_path[data].subtasks
+    assert task_result_by_path[data].subtasks[0] == task_result_by_path[out.parent].task
+    chunk_reports = [
+        report
+        for report in task_result_by_path[data].reports
+        if isinstance(report, ChunkReport)
+    ]
+    assert chunk_reports
+    assert task_result_by_path[out.parent].task.extractable_id == chunk_reports[0].id
+
+
+def test_multi_file_content_not_processed(
+    multi_volume_zip: Path,
+    multi_file_extraction_config: ExtractionConfig,
+    extraction_root: Path,
+):
+    process_result = process_file(multi_file_extraction_config, multi_volume_zip)
+    task_result_by_path = {r.task.path: r for r in process_result.results}
+
+    out = extraction_root / "input_extract" / "test.part0"
+    extract_dir = extraction_root / "input_extract" / "test.part0_extract"
+
+    assert not extract_dir.exists()
+    assert not task_result_by_path[out].subtasks
+
+
+def test_multi_file_paths_have_reports(
+    multi_volume_zip: Path,
+    multi_file_extraction_config: ExtractionConfig,
+    extraction_root: Path,
+):
+    process_result = process_file(multi_file_extraction_config, multi_volume_zip)
+    task_result_by_path = {r.task.path: r for r in process_result.results}
+
+    parts = [
+        extraction_root / "input_extract" / "test.part0",
+        extraction_root / "input_extract" / "test.part1",
+        extraction_root / "input_extract" / "test.part2",
+    ]
+
+    for part in parts:
+        assert task_result_by_path[part].reports
+        assert any(
+            isinstance(report, HashReport)
+            for report in task_result_by_path[part].reports
+        )
+        assert any(
+            isinstance(report, StatReport)
+            for report in task_result_by_path[part].reports
+        )
+        assert any(
+            isinstance(report, FileMagicReport)
+            for report in task_result_by_path[part].reports
+        )
+
+
+def test_multi_file_no_extraction(
+    multi_volume_zip: Path,
+    multi_file_extraction_config: ExtractionConfig,
+    extraction_root: Path,
+):
+    multi_file_extraction_config.dir_handlers = (NoExtractionSplitDirHandler,)
+    process_result = process_file(multi_file_extraction_config, multi_volume_zip)
+    task_result_by_path = {r.task.path: r for r in process_result.results}
+
+    directory = extraction_root / "input_extract"
+    extract_dir = extraction_root / "input_extract" / "test_extract"
+
+    assert not extract_dir.exists()
```

Comment:
original files are never removed

Reply:
this is in contradiction with the current approach with single files, right ? Files that are part of a multi-files volume should only be kept if the `-k` option is used, otherwise they must be deleted upon successful extraction/transformation.